packages/next/next.config.js

```diff
@@ -0,0 +1,39 @@
+/** @type {import('next').NextConfig} */
+const path = require("path");
+
+// Necessary to handle statically imported images
+const withImages = require('next-images');
+// Necessary to handle statically imported fonts
+const withFonts = require('next-fonts');
+
+module.exports = withImages(withFonts({
+  // Allows us to access other directories in the monorepo
+  experimental: {
+    externalDir: true
+  },
+  // This feature conflicts with next-images
+  images: {
+    disableStaticImages: true,
+  },
+  webpack: (config, options) => {
+    config.resolve.alias = {
+      ...(config.resolve.alias || {}),
+      // Transform all direct `react-native` imports to `react-native-web`
+      'react-native$': 'react-native-web',
+    }
+    config.resolve.extensions = [
+      '.web.js',
+      '.web.ts',
+      '.web.tsx',
+      ...config.resolve.extensions,
+    ]
+
+    if (options.isServer) {
+      config.externals = ['react', 'react-native-web', ...config.externals];
+    }
+    config.resolve.alias['react'] = path.resolve(__dirname, '.', 'node_modules', 'react');
+    config.resolve.alias['react-native-web'] = path.resolve(__dirname, '.', 'node_modules', 'react-native-web');
```

Comment:
I'm curious, have you already tried using `react-native-monorepo-tools`'s `addNohoistAliases` to take care of this flow? 
Otherwise, I'll give it a try later (in a separate PR) 👍

Reply:
I tried it the other week and didn't get it working, but that might've been because I had other things misconfigured. I'm AFK this weekend if you want to give this and the other changes a shot 👍🏽.